draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -385,6 +385,59 @@ in this section.
 {{examples}} contains example ML-DSA private keys encoded using the
 textual encoding defined in {{RFC7468}}.
 
+# Pre-hashed mode (ExtarnalMu-ML-DSA)
+
+
+Many applications will require a "pre-hashed" mode of ML-DSA whereby a message digest can be pre-computed outside of the cryptographic module and then only a small fixed-width hash value is passed to the crypto module.
+Many applications and protocols include message digesting, but there exist some that do not. Examples of this can be found even within [RFC5280]; for example certificate and certificate revocation list (CRL) data structures do not include message digesting and therefore become problematic when producing large CRLs or when signing a high volume of certificates containing large public keys. Such situations require pre-hashing to be performed by the signature primitive.
+
+
+This section presents the "ExternalMu-ML-DSA" signature mode which provides an interface for performing pre-hashed signatures in a way which is both compliant with [FIPS204] and which produces signatures values which are indistinguishable from signatures produced by ML-DSA.Sign() and are therefore compatible with the normal ML-DSA.Verify() and are identified by the same Object Identifiers as for ML-DSA. A ML-DSA key and certificate MAY be used with either ML-DSA or ExternalMu-ML-DSA interchangeably. Note that ExternalMu-ML-DSA describes a different signature API from ML-DSA and therefore might require explicit support from hardware or software cryptographic modules.
```

Comment:
You say that you present ExternalMu-ML-DSA, and that ExternalMu-ML-DSA really produce signatures that are indistinguishable from signatures produced by ML-DSA.Sign(). Before I go off on a tangent for no reason, are you referring to the function ExternalMu-ML-DSA defined below? or the combination of the two functions defined below?
If the former, then I disagree.  But if you mean the latter, then it makes sense.

Reply:
Yeah, I mean this entire flow. I'll re-phrase that.